proto/toit/model/data.proto

```diff
@@ -10,51 +10,60 @@ option java_package = "io.toit.proto.toit.model";
 option java_outer_classname = "DataProto";
 
 message MetricsData {
-    repeated bytes names = 1;
-    repeated GuageMetricData gauges = 2;
-    repeated CounterMetricData counters = 3;
-    repeated PlotMetricData plots = 4;
+  repeated bytes names = 1;
+  repeated GuageMetricData gauges = 2;
+  repeated CounterMetricData counters = 3;
+  repeated PlotMetricData plots = 4 [ deprecated = true ];
+  repeated HistogramMetricData histograms = 5;
 }
 
 message GuageMetricData {
-    double value = 1;
-    // name_index references the name in the names list in MetricsData.
-    uint32 name_index = 2;
-    google.protobuf.Timestamp created = 3 [deprecated=true];
-    map<uint32,uint32> tags = 4;
+  double value = 1;
+  // name_index references the name in the names list in MetricsData.
+  uint32 name_index = 2;
+  google.protobuf.Timestamp created = 3 [ deprecated = true ];
+  map<uint32, uint32> tags = 4;
 }
 
 message CounterMetricData {
-    // name_index references the name in the names list in MetricsData.
-    uint32 name_index = 1;
-    int64 count = 2;
-    double mean = 3 [deprecated=true];
-    double stdev = 4 [deprecated=true];
-    map<uint32,uint32> tags = 5;
+  // name_index references the name in the names list in MetricsData.
+  uint32 name_index = 1;
+  int64 count = 2;
+  double mean = 3 [ deprecated = true ];
+  double stdev = 4 [ deprecated = true ];
+  map<uint32, uint32> tags = 5;
 }
 
 message PlotMetricData {
-    double value = 1;
-    // name_index references the name in the names list in MetricsData.
-    uint32 name_index = 2;
-    google.protobuf.Timestamp created = 3;
-    map<uint32,uint32> tags = 4;
+  // Deprecated - use unsampled gauges instead.
+  option deprecated = true;
+  double value = 1;
+  // name_index references the name in the names list in MetricsData.
+  uint32 name_index = 2;
+  google.protobuf.Timestamp created = 3;
+  map<uint32, uint32> tags = 4;
+}
+
+message HistogramMetricData {
+  uint32 name_index = 1;
```

Comment:
Why is `name_index` different for the different metrics?

Reply:
many metrics can be sent in the same packet.